examples/src/keyring/raw_rsa/public_private_key_separate.py

```diff
@@ -0,0 +1,122 @@
+    """
+    # Prepare your encryption context.
+    # https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
+    encryption_context = {
+        "encryption": "context",
+        "is not": "secret",
+        "but adds": "useful metadata",
+        "that can help you": "be confident that",
+        "the data you are handling": "is what you think it is",
+    }
+
+    # Generate an RSA private key to use with your keyring.
+    # In practice, you should get this key from a secure key management system.
+    #
+    # Why did we use this public exponent?
+    # https://latacora.singles/2018/04/03/cryptographic-right-answers.html
+    private_key = rsa.generate_private_key(public_exponent=65537, key_size=4096, backend=default_backend())
+
+    # Collect the public key from the private key.
+    public_key = private_key.public_key()
+
+    # The keyring determines how your data keys are protected.
+    #
+    # Create the encrypt keyring that only has access to the public key.
+    public_key_keyring = RawRSAKeyring(
+        # The key namespace and key name are defined by you
+        # and are used by the raw RSA keyring
+        # to determine whether it should attempt to decrypt
+        # an encrypted data key.
+        #
+        # https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html#use-raw-rsa-keyring
+        key_namespace="some managed raw keys",
+        key_name=b"my RSA wrapping key",
+        public_wrapping_key=public_key,
+        # The wrapping algorithm tells the raw RSA keyring
+        # how to use your wrapping key to encrypt data keys.
+        #
+        # Why did we use this wrapping algorithm?
+        # https://latacora.singles/2018/04/03/cryptographic-right-answers.html
+        wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+    )
+
+    # Create the decrypt keyring that has access to the private key.
+    private_key_keyring = RawRSAKeyring(
+        # The key namespace and key name MUST match the encrypt keyring.
+        key_namespace="some managed raw keys",
+        key_name=b"my RSA wrapping key",
+        private_wrapping_key=private_key,
+        # The wrapping algorithm MUST match the encrypt keyring.
+        wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+    )
+
+    # Encrypt your plaintext data using the encrypt keyring.
+    ciphertext, _encrypt_header = aws_encryption_sdk.encrypt(
+        source=source_plaintext, encryption_context=encryption_context, keyring=public_key_keyring
+    )
+
+    # Verify that the ciphertext and plaintext are different.
+    assert ciphertext != source_plaintext
+
+    # Decrypt your encrypted data using the decrypt keyring.
+    #
+    # We do not need to specify the encryption context on decrypt
+    # because the header message includes the encryption context.
+    decrypted, decrypt_header = aws_encryption_sdk.decrypt(source=ciphertext, keyring=private_key_keyring)
```

Comment:
This should probably be moved under the exception handling case so the flow is..
1. Encrypt
2. Try decrypt for public key keyring, which fails
3. Try decrypt with private key keyring, which succeeds

Rather than 
1. Encrypt
2. Decrypt with private key keyring
3. Try decrypt for public key keyring, which fails
4. Verify private key keyring decryption was correct